mem/index.d.ts

```diff
@@ -0,0 +1,18 @@
+// Type definitions for mem 1.1.0
+// Project: https://github.com/sindresorhus/mem
+// Definitions by: Sam Verschueren <https://github.com/SamVerschueren>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare module "mem" {
+	interface Options {
+		maxAge?: number;
+		cacheKey?: (...args: any[]) => string;
+		cache?: any;
+	}
+
+	function mem<T>(fn: (...args: any[]) => T, options?: Options): (...args: any[]) => T;
```

Comment:
Good point, not sure what would fit the best. What your declaration, you would have to write something like this.

```js
const memGot = mem<(url: string) => Promise<{body: any}>>(got);
```

With my declaration, it looks like this

```js
const memGot = mem<Promise<{body: any}>>(got);
```

But I don't have control over the parameters of the function while with yours, you have. So yes, yours might be slightly better.